staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/stdlib.go

```diff
@@ -0,0 +1,634 @@
+			decls.NewListType(decls.String)),
+	),
+	decls.NewFunction("url",
+		decls.NewOverload("string_to_url",
+			[]*exprpb.Type{decls.String},
+			model.URLObject),
+	),
+	decls.NewFunction("getScheme",
+		decls.NewInstanceOverload("url_get_scheme_string",
+			[]*exprpb.Type{model.URLObject},
+			decls.String),
+	),
+	decls.NewFunction("getHost",
+		decls.NewInstanceOverload("url_get_host_string",
+			[]*exprpb.Type{model.URLObject},
+			decls.String),
+	),
+	decls.NewFunction("getHostname",
+		decls.NewInstanceOverload("url_get_hostname_string",
+			[]*exprpb.Type{model.URLObject},
+			decls.String),
+	),
+	decls.NewFunction("getPort",
+		decls.NewInstanceOverload("url_get_port_string",
+			[]*exprpb.Type{model.URLObject},
+			decls.String),
+	),
+	decls.NewFunction("getEscapedPath",
+		decls.NewInstanceOverload("url_get_escaped_path_string",
+			[]*exprpb.Type{model.URLObject},
+			decls.String),
+	),
+	decls.NewFunction("getQuery",
+		decls.NewInstanceOverload("url_get_query_string",
+			[]*exprpb.Type{model.URLObject},
+			decls.NewMapType(decls.String, decls.NewListType(decls.String))),
+	),
+	decls.NewFunction("isURL",
+		decls.NewOverload("string_is_url_bool",
+			[]*exprpb.Type{decls.String},
+			decls.Bool),
+	),
+}
+
+func (stdlib) CompileOptions() []cel.EnvOption {
+	return []cel.EnvOption{
+		cel.Declarations(stdlibDecls...),
+	}
+}
+
+func (stdlib) ProgramOptions() []cel.ProgramOption {
+	return []cel.ProgramOption{
+		cel.Functions(
+			&functions.Overload{
+				Operator: "isSorted",
+				Unary:    isSorted,
+			},
+			// use overload names for sum so an initial accumulator value can be assigned to each
+			&functions.Overload{
+				Operator: "list_int_sum_int",
+				Unary: sum(func() ref.Val {
+					return types.Int(0)
+				}),
+			},
+			&functions.Overload{
+				Operator: "list_uint_sum_uint",
+				Unary: sum(func() ref.Val {
+					return types.Uint(0)
+				}),
+			},
+			&functions.Overload{
+				Operator: "list_double_sum_double",
+				Unary: sum(func() ref.Val {
+					return types.Double(0.0)
+				}),
+			},
+			&functions.Overload{
+				Operator: "list_duration_sum_duration",
+				Unary: sum(func() ref.Val {
+					return types.Duration{Duration: 0}
+				}),
+			},
+			&functions.Overload{
+				Operator: "max",
+				Unary:    max(),
+			},
+			&functions.Overload{
+				Operator: "min",
+				Unary:    min(),
+			},
+			// use overload names for indexOf and lastIndexOf to de-conflict with function of same name in strings extension library
+			&functions.Overload{
+				Operator: "list_a_index_of_int",
+				Binary:   indexOf,
+			},
+			&functions.Overload{
+				Operator: "list_a_last_index_of_int",
+				Binary:   lastIndexOf,
+			},
+			&functions.Overload{
+				Operator: "find",
+				Binary:   find,
+			},
+			&functions.Overload{
+				Operator: "findAll",
+				Binary: func(str, regex ref.Val) ref.Val {
+					return findAll(str, regex, types.Int(-1))
+				},
+				Function: findAll,
+			},
+			&functions.Overload{
```

Comment:
I've done this everywhere I think I can.  I can't do it for indexOf/lastIndexOf because the function declaration is part of strings.go in CEL, and I don't think I can do it for sum because I don't have an implementation that works if I don't know the type (since the accumulator needs to be initialized to 0 for the correct numeric type).

Reply:
You can make the dynamic dispatch work if you can inspect the first element in the list, but when the list is empty in the dynamic dispatch case, you could just error. Just a thought.